ehrql/main.py

```diff
@@ -56,6 +57,10 @@ def generate_dataset(
         definition_file, user_args, environ
     )
 
+    if test_data_file:
+        log.info(f"Testing dataset definition with tests in {str(definition_file)}")
+        assure(test_data_file, environ=environ, user_args=user_args)
```

Comment:
It looks like the `assure` command never actually triggers an error: it just prints some stuff and exists successfully regardless of the whether the tests passed or failed:
https://github.com/opensafely-core/ehrql/blob/e863a3409c6c0473109dbfe3d1567c50eacb9b30/ehrql/main.py#L323-L328

I would have expected the command to fail if the test failed.

---

Update: I've just read this:

> For now, this will print a warning if the tests fail. We should think about whether we want the command to completely fail when this happens.